exonum/src/api/node/public/explorer.rs

```diff
@@ -171,6 +171,35 @@ impl AsRef<[u8]> for TransactionHex {
     }
 }
 
+/// Status of a call.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct CallStatusResponse {
+    /// Call status
+    pub status: Result<(), ExecutionError>,
+    /// Call execution proff
+    pub call_proof: MapProof<CallInBlock, ExecutionError>,
+}
+
+/// The kind of requested call
+#[derive(Debug, Clone, Copy, Eq, Serialize, Deserialize, PartialEq, Hash)]
+pub enum CallKind {
+    /// Call kind of transaction
+    Transaction,
+    /// Call kind of `before_transactions` hook
+    BeforeTransactions,
+    /// Call kind of `after_transactions` hook
+    AfterTransactions,
+}
+
+/// Call status query parameters.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq)]
+pub struct CallStatusQuery {
```

Comment:
I expect the query to serialize to something like

```text
height=10&type=transaction&index=3
```

or

```text
height=10&type=after_transactions&instance_id=5
```

I think the easiest way to accomplish this is by using

```rust
struct CallStatusQuery {
    height: Height, // or u64
    #[serde(flatten)]
    call: CallInBlock,
}
```

If we want to support searching by `tx_hash` too (fair enough), then it could be accomplished with

```rust
#[serde(untagged)]
enum CallStatusQuery {
    TransactionHash { tx_hash: Hash },
    CallWithHeight {
        height: Height, // or u64
        #[serde(flatten)]
        call: CallInBlock,
    },
}
```